### PR TITLE
New pytorch compatibility

### PR DESCRIPTION
**Pull request by schrum2**
Hi @TheHedgeify , This is potentially a big change, so I wanted to submit it as a pull request rather than just merge it without letting you know.

It turns out that sometime after we started working on Mario GAN, Pytorch got upgraded and no longer allows "." in the names of network layers. This means that anyone who discovers our repo and simply installs the latest version of PyTorch in order to test our code won't be able to run it. I just experienced this myself since I installed Pytorch on a different machine to run this code.

So, I made changes that should allow the code to train a new GAN with the latest versions of PyTorch (though I did not test this) and also added code that re-labels the network layer names of old trained models, so that the models you already trained can be successfully loaded. This is something I did test by running evaluateLevel.

I also made some tweaks that allow CMAMarioSolver to run with the old MarioGAN that Sebastian trained for GECCO 2018, though technically this is using your new LevelParser and not the old one (not sure if this is an actual problem ... maybe the new tile types simply are not there).

So, after you test this out and make sure it doesn't cause any problems with the experiments you are running, you can merge it into gbea branch. However, we may also be able to merge this into master branch as well (that will be a separate pull request).

Let me know what you think.

### CODE REVIEW

**Comment by TheHedgeify** [comment]
Hi! Thanks for looking into the python version. I had similar issues and just decided to downgrade python. GAN training still works for me.

However, the generator_ws.py seems to have a small issue due to the update:
pytorch/generator_ws.py:123: UserWarning: volatile was removed and now has no effect. Use `with torch.no_grad():` instead.
  levels = generator(Variable(latent_vector, volatile=True))

Using the new LevelParser with the old encoding will not work out-of-the-box, there are too many changes. Neither should loading the old .pth files with the new Level Parser. It will put whole pipes where there should only be one tile of them and the order of the tiles in the encoding probably changed...

**Comment by schrum2** [comment]
I don't think the ordering of the tiles got too messed up, since the levels that are generated seem to look ok. However, the pipes are definitely a difference. I think the easy solution is to simply copy the old LevelParser from master branch into OldLevelParser and then have some sort of conditional switch that uses one or the other.
